client/platform/desktop/frontend/store/settings.ts

```diff
@@ -30,30 +30,38 @@ function isSettings(s: any): s is Settings {
   return true;
 }
 
+// Settings initialization involves a handoff between renderer and background.
+// This function should be called at application startup.
 async function init() {
-  const settingsStr = window.localStorage.getItem(SettingsKey);
-  let settingsvalue = await getDefaultSettings();
+  // Client asks for default settings and external force overrides from background
+  let settingsValue = await getDefaultSettings();
   try {
-    if (settingsStr) {
-      const maybeSettings = JSON.parse(settingsStr);
-      if (isSettings(maybeSettings)) {
-        settingsvalue = {
-          // Populate from defaults to include any missing properties
-          ...settingsvalue,
-          // Overwrite with explicitly persisted settings
-          ...maybeSettings,
-        };
-      }
+    // Client applies user-configured settings from localstorage
+    const settingsStr = window.localStorage.getItem(SettingsKey) || '{}';
+    const maybeSettings = JSON.parse(settingsStr);
+    if (isSettings(maybeSettings)) {
+      settingsValue = {
+        // Populate from defaults to include any missing properties
+        ...settingsValue,
+        // Overwrite with explicitly persisted settings
+        ...maybeSettings,
+      };
     }
   } catch {
     // pass
   }
-  settings.value = settingsvalue;
-
+  // Client applies external force overrides
+  if (settingsValue.overrides.viamePath !== undefined) {
+    settingsValue.viamePath = settingsValue.overrides.viamePath;
+  }
+  if (settingsValue.overrides.readonlyMode !== undefined) {
+    settingsValue.readonlyMode = settingsValue.overrides.readonlyMode;
+  }
+  settings.value = settingsValue;
   ipcRenderer.send('update-settings', settings.value);
```

Comment:
I don't know exactly if this would be a incorrect/correct option but if the settings are different than the base `localSettings` like the case where there is an environment variable.  Here is where I think you would want to just call `updateSettings` from further down to update the `localStorage` as well